packages/components/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  createRef,
+  useEffect,
+  useState,
+} from "react";
+import classnames from "classnames";
+import ReactDOM from "react-dom";
+import { AnimatePresence, motion } from "framer-motion";
+import { Text } from "../Text";
+import styles from "./Tooltip.css";
+
+interface TooltipProps {
+  readonly children: ReactElement;
+  /**
+   * Tooltip text
+   */
+  readonly message: string;
+}
+
+export function Tooltip({ message, children }: TooltipProps) {
+  const [direction, setDirection] = useState("above");
+  const [showOnLoad, setShowOnLoad] = useState(false);
+  const [position, setPosition] = useState({ top: "0px", left: "0px" });
+  const tooltipRef = createRef<HTMLDivElement>();
+  const shadowRef = createRef<HTMLSpanElement>();
+
+  showOnHover(shadowRef, setShowOnLoad);
+
+  useEffect(() => {
+    if (
+      tooltipRef.current &&
+      shadowRef.current &&
+      shadowRef.current.nextElementSibling
+    ) {
+      const activator = shadowRef.current.nextElementSibling;
+      const tooltip = tooltipRef.current;
+      const activatorBounds = activator.getBoundingClientRect();
+      const xOffset =
```

Comment:
I feel like it might be nice to extract some of this positioning logic out into a function. Perhaps something like…

```ts
getPosition(direction: "above" | "below", activator: HTMLElement, tooltip: HTMLElement): {x: number, y: number};
```

Reply:
would be nice. lets see if i can get that working without some crazy react types